Description: fix configuration path (and crash on startup)
Origin: https://bitbucket.org/tortoisehg/thg/changeset/9635693f3b973702fdee4e8fa487afb4d93959be
Bug-Debian: http://bugs.debian.org/678558

# HG changeset patch
# User Jairo Llopis <yajo.sk8@gmail.com>
# Date 1342020308 -7200
# Node ID 9635693f3b973702fdee4e8fa487afb4d93959be
# Parent  b73053f7cacb5ca8679d09b81363cd62473f5496
nautilus: Fix issue #869 [1]. Nautilus extension now works.

The nautilus extension now uses $XDG_CONFIG_HOME like the rest of thg.

Links:
    [1] https://bitbucket.org/tortoisehg/thg/issue/869/tortoisehg-nautilus-doesnt-work-at-all

Index: tortoisehg-2.4/contrib/nautilus-thg.py
===================================================================
--- tortoisehg-2.4.orig/contrib/nautilus-thg.py	2012-09-16 22:35:49.904065810 +0200
+++ tortoisehg-2.4/contrib/nautilus-thg.py	2012-09-16 22:36:23.272376213 +0200
@@ -66,10 +66,25 @@
         from tortoisehg.util import menuthg
         self.hgtk = paths.find_in_path(thg_main)
         self.menu = menuthg.menuThg()
-        self.notify = os.path.expanduser('~/.tortoisehg/notify')
 
-        f = open(self.notify, 'w')
-        f.close()
+        # Get the configuration directory path
+        try:
+            self.notify = os.environ['XDG_CONFIG_HOME']
+        except KeyError:
+            self.notify = os.path.join('$HOME', '.config')
+
+        self.notify = os.path.expandvars(os.path.join(
+            self.notify,
+            'TortoiseHg'))
+
+        # Create folder if it does not exist
+        if not os.path.isdir(self.notify):
+            os.makedirs(self.notify)
+
+        # Create the notify file
+        self.notify = os.path.join(self.notify, 'notify')
+        open(self.notify, 'w').close()
+
         self.gmon = Gio.file_new_for_path(self.notify).monitor(Gio.FileMonitorFlags.NONE, None)
         self.gmon.connect('changed', self.notified)
 
